File: Topics/Recursion/GdeleteMiddleElementStack.java
import java.io.*;
import java.util.*;

public class GdeleteMiddleElementStack {

	static Stack<Integer> deleteMiddleElement(Stack<Integer> s, int target){

		if(target == 1){
			s.pop();
			return s;			
		}

		int val = s.peek();
		s.pop();
		deleteMiddleElement(s, target - 1);
		s.push(val);
		return s;
	}

	// Utility Method to print contents of stack
	static void printStack(Stack<Integer> s) {
		ListIterator<Integer> lt = s.listIterator();

		// forwarding
		while (lt.hasNext())
			lt.next();

		// printing from top to bottom
		while (lt.hasPrevious())
			System.out.print(lt.previous() + " ");
	}

	public static void main (String[] args) {

		try {
			System.setIn(new FileInputStream("input.txt"));
			System.setOut(new PrintStream(new FileOutputStream("output.txt")));
		} catch (Exception e) {
			System.err.println("Error");
		}
	    
	    //taking input using Scanner class
		Scanner sc=new Scanner(System.in);
		
		//taking total testcases
		int T=sc.nextInt();
		while(T-->0)
		{
		 	int size = sc.nextInt();
		 	Stack<Integer> st = new Stack();

		 	for (int i = 0; i < size; i++) {
		 		int val = sc.nextInt();
		 		st.push(val);
		 	}

		 	int target = (size/2) + 1;
		 	deleteMiddleElement(st, target);
		 	printStack(st);
		}
		
	}
}

File: Topics/Recursion/GdeleteMiddleElementStack.cpp
#include <bits/stdc++.h>
using namespace std;



/*

	Middle element:
	I/P -> 1 2 3 4 5	O/P -> 1 2 4 5
	I/P -> 1 2 3 4 5 6	O/P -> 1 2 3 5 6

	Middle Element:
	(size/2) + 1
*/

stack<int> deleteMiddleElement(stack<int> &nums, int targetIndex) {

	if (targetIndex == 1)
	{
		nums.pop();
		return nums;
	}

	int val = nums.top();
	nums.pop();
	nums = deleteMiddleElement(nums, targetIndex-1);
	nums.push(val);
	return nums;
}



int main() {

#ifndef ONLINE_JUDGE
	// for getting input from input.txt
	freopen("input.txt", "r", stdin);
	// for writing output to output.txt
	freopen("output.txt", "w", stdout);
#endif
	int size;
	cin >> size;
	stack<int> nums;
	for (int i = 0; i < size; i++)
	{
		int x; cin >> x;
		nums.push(x);
	}
	int target = (size / 2) + 1;
	nums = deleteMiddleElement(nums, target);
	while (!nums.empty()) {

		int val = nums.top();
		cout << val << " ";
		nums.pop();
	}

}